Realm/ObjectServerTests/RLMAsymmetricSyncServerTests.mm

```diff
@@ -0,0 +1,163 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import "RLMSyncTestCase.h"
+
+// We add this compilation flag, because we don't want to compile this code when we run
+// `xcrun swift test`. Adding an asymmetric table to a local realm will throw a fatal error
+// when opening the realm, excluding this from compilation will not add the object to
+// the schema. We are only running this tests on the ObjectServer target.
+#if REALM_SERVER_TEST_TARGET
```

Comment:
We need to run all of the tests via SPM as that's the only way for us to test with asan and tsan. If the problem is that the type ends up in the default schema, the class should just override the class function to exclude it.

Reply:
Changed this completely so instead of using a build configuration flag,  I'm creating an app exclusively for asymmetric testing which is not using the sharedSchema to create the app.